app/celery/letters_pdf_tasks.py

```diff
@@ -262,8 +263,20 @@ def shatter_deliver_letter_tasks(notification_ids):
     # If the number or size of arguments to this function change, then the default
     # `batch_size` argument of `send_dvla_letters_via_api` needs updating to keep
     # within SQS’s maximum message size
+    queue_name = QueueNames.SEND_LETTER
+
     for id in notification_ids:
-        deliver_letter.apply_async(kwargs={"notification_id": id}, queue=QueueNames.SEND_LETTER)
+        message_group_kwargs = {}
+        if (current_app.config.get("ENABLE_SQS_FAIR_GROUPING", False)):
+            notification = get_notification_by_id(id)
```

Comment:
This celery task was explicitly written so it didn't have to hit the database and this perfectly illustrates my objection to implementing group ids in this way, as mentioned here https://github.com/alphagov/notifications-api/pull/4676#issuecomment-3701964556 and in many other places.